downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -45,7 +45,7 @@ import org.ossreviewtoolkit.downloader.WorkingTree
 import org.ossreviewtoolkit.model.VcsInfo
 import org.ossreviewtoolkit.model.VcsType
 import org.ossreviewtoolkit.utils.CommandLineTool
-import org.ossreviewtoolkit.utils.LicenseFilenamePatterns
+import org.ossreviewtoolkit.utils.LicenseFilenamePatterns.ALL_LICENSE_FILENAMES
```

Comment:
> While this behavior is correct it is also ok to checkout all files
matching any license filename patterns irrespective of the directory.

I don't get *why* this is ok. If we also checkout license files from any directory, these files get scanned by ScanCode, and findings irrelevant to the VCS path get reported. This would only be ok if we would filter ScanCode results to the VCS path, which AFAIK is something that we do not do yet.

Reply:
I think it is ok, because we do actually filter the scan results by VCS path.

Reply:
I must have missed this being added then. Can you please point me at the code location doing this? Because if that part is already there, it should be very easy to indeed stop doing sparse checkouts. So far, the lack of such scan result filtering by path was the blocker to drop sparse checkouts, IMO.

Reply:
Sure, please see
- https://github.com/oss-review-toolkit/ort/blob/master/scanner/src/main/kotlin/LocalScanner.kt#L558-L562
- https://github.com/oss-review-toolkit/ort/blob/master/scanner/src/main/kotlin/Scanner.kt#L152-L168

Reply:
I think this is the PR that introduced the change:
https://github.com/oss-review-toolkit/ort/pull/3258

Reply:
The commit messages should say why it is ok to checkout those files, otherwise it's a bit confusing.

Reply:
Done.